source/1_engine/rex_engine/src/diagnostics/assert.cpp

```diff
@@ -0,0 +1,85 @@
+#include "rex_engine/diagnostics/assert.h"
+
+#include "rex_engine/debug_types.h"
+#include "rex_engine/diagnostics/logging/log_macros.h"
+#include "rex_engine/diagnostics/stacktrace.h"
+
+namespace rex
+{
+  DECLARE_LOG_CATEGORY_EXTERN(LogAssert, LogVerbosity::Log);
+  DEFINE_LOG_CATEGORY(LogAssert);
+
+  DebugVector<AssertContext>& contexts()
+  {
+    thread_local DebugVector<AssertContext> contexts;
+    return contexts;
+  }
+
+  void rex_assert(const rsl::fmt_stack_string& msg)
+  {
+    thread_local static bool is_processing_assert = false;
```

Comment:
Why does this need to be thread local?

Reply:
because different threads could be processing an assert at the exact same time